src/webui/www/private/scripts/client.js

```diff
@@ -781,8 +781,26 @@ window.addEvent('load', function() {
         addMainWindowTabsEventListener();
         addSearchPanel();
     }
+
+    registerMagnetHandler();
+    handleMagnetParam();
 });
 
+function registerMagnetHandler() {
+    const urlObject = new URL(location.href);
+    urlObject.searchParams.set('magnet', '');
+    const templateUrl = urlObject.href.replace('magnet=', 'magnet=%s');
+    navigator.registerProtocolHandler('magnet', templateUrl, 'Magnet Handler');
```

Comment:
I'm getting `TypeError: navigator.registerProtocolHandler is not a function` error over unencrypted http connection, perhaps you should return early at the top of this function if it is not using https?

Reply:
Ah, I wasn't aware of this. Will ensure the function exists before trying to call it.